### PR TITLE
Add testcase for checking the stream logic in logCollector

### PR DESCRIPTION
**Pull request by Niveditha-phData**
- Added TestStream class in SolrCloudStreamTest.scala
- Added two test cases to verify whether flush happens after n events and n seconds

### CODE REVIEW

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala [hunk 1]
```diff
@@ -118,4 +120,28 @@ class SolrCloudStreamTest extends FunSuite with BaseSolrCloudTest {
...
+    val stream = new TestStream(1, 10)
+
+    (1 to 3).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 3)
+  }
+}
+
+class TestStream(flushDuration: Int, flushSize: Int) extends Stream[String](StreamParams()) {
+  StreamParams.apply(1, 100, flushSize, flushDuration, "fail")
```
Comment (afoerster) [inline log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: No need to call `apply` here, you can just call `StreamParams(args)`

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala [hunk 2]
```diff
@@ -118,4 +120,28 @@ class SolrCloudStreamTest extends FunSuite with BaseSolrCloudTest {
...
+  test("should flush after n seconds") {
+    val stream = new TestStream(1, 10)
+
+    (1 to 3).foreach(x => stream.put("testApp", x.toString))
+    Thread.sleep(2500)
+    assert(stream.results.length === 3)
+  }
+}
+
+class TestStream(flushDuration: Int, flushSize: Int) extends Stream[String](StreamParams()) {
```
Comment (afoerster) [inline log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: Let's put the `Duration` into the argument. You can do `import scala.concurrent.duration._`, then create a duration like `1 second`. This also disambiguates the two arguments, it's easy to make errors when you have two int arguments next to eachother, it's also harder to read

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala [hunk 3]
```diff
@@ -21,6 +21,8 @@ import io.phdata.pulse.solr.{ BaseSolrCloudTest, TestUtil }
 import org.apache.solr.client.solrj.SolrQuery
 import org.scalatest.FunSuite
 
+import scala.collection.mutable
+
 class SolrCloudStreamTest extends FunSuite with BaseSolrCloudTest {
   val TEST_COLLECTION = TestUtil.randomIdentifier()
```
Comment (afoerster) [inline log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: Is this still used?

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala [hunk 4]
```diff
@@ -21,6 +21,8 @@ import io.phdata.pulse.solr.{ BaseSolrCloudTest, TestUtil }
 import org.apache.solr.client.solrj.SolrQuery
```
Comment (afoerster) [inline log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: Make sure to remove all unused imports

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 3 above)
Comment (Niveditha-phData) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: I am pushing the test events in mutable.seq

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 4 above)
Comment (Niveditha-phData) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: I have added the testcases in the existing test suite. `SolrQuery` is already used in line-64 and 87.

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 2 above)
Comment (Niveditha-phData) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: StreamParams takes flushDuration as Int. So the argument is kept  as Int. Direct conversion from Duration to Int is not there. Can I keep it like this or I try ways for this conversion

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 1 above)
Comment (Niveditha-phData) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: I will do this.

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 4 above)
Comment (afoerster) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: oh you need to delete the ignored tests

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 3 above)
Comment (afoerster) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: I meant `TEST_COLLECTION`

File: log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala (hunk 2 above)
Comment (afoerster) [reply log-collector/src/test/scala/io/phdata/pulse/logcollector/SolrCloudStreamTest.scala]: let's leave it the way it is for now

**Pull request merged**